### DISCUSSION TITLE
shouldn't `OWLVarDecl *vars` be const?

### DISCUSSION THREAD

**Discussion by bricerebsamen** [owl-project/owl#207] [General]
In `owlGeomTypeCreate()` https://github.com/owl-project/owl/blob/056e8ffc68bf236006e24ea6a82ed6f38d4861bd/owl/impl.cpp#L903 and similar functions in this file, shouldn't shouldn't `OWLVarDecl *vars` be const since it's passed to `checkAndPackVariables` which takes it as a const?

Also, does this array need to remain valid after the call to `owlGeomTypeCreate()` or can it be a temp variable? for now I have it as a static, which means that it should also be const...

**Comment by ingowald** [comment]
This is correct, the could (and arguably should) be declared const.
They will not be modified, and will get copied when the type gets created, so can absoltely live in temp stack mem - or be free'd right after use.

**Comment by bricerebsamen** [comment]
PR #208
